lib/slots.ts

```diff
@@ -287,5 +303,15 @@ export function SetSlot<KeyT extends SlotKey>(
   id: KeyT,
   value: Slots[KeyT]['value']
 ): void {
-  GetSlots(container)[id] = value;
+  const slots = GetSlots(container);
+
+  if (slots === undefined) {
+    throw new TypeError('Missing slots for the given container');
+  }
+
+  if (id in slots) {
```

Comment:
Wouldn't it be better to write this as 

const existingSlot = slots[id];
if (existingSlot) throw new TypeError(...);
slots[id] = value;

Reply:
Fixed it as suggested!